project1/server/agent.cpp

```diff
@@ -4,231 +4,415 @@
+                perror("Failed to read from client socket");
+                parser_.ResetParser();
+                return ClientState::ERROR;
+            } else if (bytes_read == 0) {
+                // Client has disconnected nicely.
+                std::cout << "Detected client disconnect." << std::endl;
+                return ClientState::DISCONNECTED;
+            }
+
+            // The parser assumes that the entire vector contains valid data, so limit
+            // the size.
+            incoming_message_buffer_.resize(bytes_read);
+
+            // Add the data to the parser.
+            parser_.AddNewData(incoming_message_buffer_);
+        }
+
+        // Get the parsed message.
+        if (!parser_.GetMessage(message)) {
+            std::cerr << "Failed to get the parsed message." << std::endl;
+            return ClientState::ERROR;
+        }
+        return ClientState::ACTIVE;
+    }
+
+    Agent::ClientState Agent::DispatchMessage(const Request &message) {
+        if (message.has_get()) {
+            return HandleRequest(message.get());
+        } else if (message.has_put()) {
+            return HandleRequest(message.put());
+        } else if (message.has_delete_()) {
+            return HandleRequest(message.delete_());
+        } else if (message.has_list()) {
+            return HandleRequest(message.list());
+        } else if (message.has_change_dir()) {
+            return HandleRequest(message.change_dir());
+        } else if (message.has_make_dir()) {
+            return HandleRequest(message.make_dir());
+        } else if (message.has_pwd()) {
+            return HandleRequest(message.pwd());
+        } else if (message.has_quit()) {
+            return HandleRequest(message.quit());
+        } else if (message.has_terminate()) {
+            return HandleRequest(message.terminate());
+        }
+
+        std::cerr << "No valid message was received." << std::endl;
+        return ClientState::ERROR;
+    }
+
+    bool Agent::SendResponse(const Response &response) {
+        // Serialize the message.
+        if (!wire_protocol::Serialize(response, &outgoing_message_buffer_)) {
+            std::cerr << "Failed to serialize message." << std::endl;
+            return false;
+        }
+
+        // Send the message.
+        if (send(client_fd_, outgoing_message_buffer_.data(),
+                 outgoing_message_buffer_.size(), 0) < 0) {
+            perror("Failed to send the message");
+            return false;
+        }
+        return true;
+    }
+
+    bool Agent::SendFileContents(const ftp_messages::FileContents &file_contents, uint16_t command_id) {
+        // Serialize the message.
+        if (!wire_protocol::Serialize(file_contents, &outgoing_message_buffer_)) {
+            std::cerr << "Failed to serialize message." << std::endl;
+            return false;
+        }
+
+        // monitor for termination request for get and put commands
+        uint16_t totalBytesSent = 0;
```

Comment:
There are a bunch of variables here that do not follow the style guide naming rules.

Reply:
These will be changed.